geonode_mapstore_client/migrations/0002_migrate_map_blob.py

```diff
@@ -0,0 +1,99 @@
+drop_mapstore2_adapter_mapstoredata = "DROP TABLE IF EXISTS mapstore2_adapter_mapstoredata CASCADE;"
+drop_mapstore2_adapter_mapstoreresource = "DROP TABLE IF EXISTS mapstore2_adapter_mapstoreresource CASCADE;"
+drop_mapstore2_adapter_mapstoreresource_attributes = "DROP TABLE IF EXISTS mapstore2_adapter_mapstoreresource_attributes CASCADE;"
+
+
+def migrate_map_forward(apps, schema_editor):
+
+    exists = False
+    sql_exists = "SELECT EXISTS (SELECT FROM information_schema.tables WHERE table_name = 'mapstore2_adapter_mapstoredata');"
+    with connections['default'].cursor() as cursor:
+        cursor.execute(sql_exists)
+        result = cursor.fetchall()
+        if result:
+            exists = result[0][0]
+    if exists:
+        # We can't import the Map model directly as it may be a newer
+        # version than this migration expects. We use the historical version.
+        ResourceBase = apps.get_model('base', 'ResourceBase')
+        for _resource in ResourceBase.objects.filter(resource_type='map').iterator():
+            to_update = {}
+            # We can't use map.resourcebase_ptr, we need to explicitly retrieve the resourcebase
+            # mapstore2_adapter does not exist anymore as an app. So we need raw sql to get the blob from the old table.
+            try:
+                sql_string = f'SELECT blob from mapstore2_adapter_mapstoredata where resource_id={_resource.id};'
+                '''
+                Getting the Data Blob
+                '''
+                with connections['default'].cursor() as cursor:
+                    cursor.execute(sql_string)
+                    result = cursor.fetchall()
+                    if result:
+                        to_update['blob'] = result[0][0]
+
+                sql_string = f'SELECT name, value from mapstore2_adapter_mapstoreattribute where resource_id={_resource.id};'
+                '''
+                Getting the attributes
+                '''
+                with connections['default'].cursor() as cursor:
+                    cursor.execute(sql_string)
+                    result = cursor.fetchall()
+                    if result:
+                        for x in result:
+                            try:
+                                '''
+                                If is a byte we have to decode it
+                                '''
+                                to_update[x[0]] = base64.b64decode(ast.literal_eval(x[1])).decode()
+                            except:
+                                to_update[x[0]] = x[1]
+
+                    thumb = to_update.pop('thumbnail', None)
+                    if thumb and 'data:image/' not in thumb:
+                        to_update['thumbnail_url'] = to_update.pop('thumbnail')
+                '''
+                Updating the resource
+                '''
+                if to_update:
+                    ResourceBase.objects.filter(id=_resource.id).update(**to_update)
+
```

Comment:
I would suggest running the `drop_mapstore2_adapter_*` SQL here, before the `Exception` and only once the data migration has successfully completed. 

Also, the exception should be focused to a specific `DoesNotExist` rather than a catch-anything exception that will prevent the user knowing anything went wrong with their data. 

This should prevent the situation where an exception occurs between line 31 and 65, it is caught in line 67 and then `continue` allows the four tables to still be dropped, via lines 95 to 98, even though the data has not been successfully moved into the `resourcebase` yet. This situation could create a very unfortunate test of your ability to restore a database from backups (!).

Reply:
You are correct, the exception was permissive. I'm going to remove it so that the migration process will stop in case of an exception (any kind). So that, in case of an exception the migration is stoped and the table are not droped
Related to the table drop, I'll prefer to keep them into another migration operation to make it more transparent and gives a single operation his job.

Reply:
OK, perfect. Yes I think that's safest. If something fails in a data migration I want it to throw an error, so I can go back & fix the data.

Thanks for your work on this!